[Python] Incorrect argument not flagged

For this program:
```python
def foo(x: int, y: str) -> None:
    pass

foo(42, x=1, y="foo")
```
ESBMC reports verification success:
```
$ esbmc ex.py 
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py

Type checking warning:
ex.py:4: error: "foo" gets multiple values for keyword argument "x"  [misc]
Found 1 error in 1 file (checked 1 source file)

Converting
Generating GOTO Program
GOTO program creation time: 0.550s
GOTO program processing time: 0.002s
Starting Bounded Model Checking
Symex completed in: 0.000s (3 assignments)
Caching time: 0.000s (removed 0 assertions)
Slicing time: 0.000s (removed 3 assignments)
Generated 0 VCC(s), 0 remaining after simplification (0 assignments)
BMC program time: 0.000s

VERIFICATION SUCCESSFUL
```
If keyword arguments are used, the error is correctly captured, e.g.:
```python
foo(z=42, x=1, y="foo")
```
```
$ esbmc ex.py 
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py

Type checking warning:
ex.py:1: note: "foo" defined here
ex.py:4: error: Unexpected keyword argument "z" for "foo"  [call-arg]
Found 1 error in 1 file (checked 1 source file)

Converting
ERROR: Unknown keyword argument: z in function foo
```
ESBMC version: 0bcfc31fe

--- a/regression/python/github_3015/main.py
+++ b/regression/python/github_3015/main.py
@@ -1,0 +1,8 @@
+# Test: Correct function call with keyword arguments (should succeed)
+# Issue #3015: Correct usage of keyword arguments
+def foo(x: int, y: str) -> None:
+    assert x == 42
+    assert y == "foo"
+
+foo(x=42, y="foo")
+

--- a/regression/python/github_3015/test.desc
+++ b/regression/python/github_3015/test.desc
@@ -1,0 +1,5 @@
+CORE
+main.py
+
+^VERIFICATION SUCCESSFUL$
+

--- a/regression/python/github_3015_1_fail/main.py
+++ b/regression/python/github_3015_1_fail/main.py
@@ -1,0 +1,7 @@
+# Test: Unknown keyword argument (should fail)
+# Issue #3015: Passing unexpected keyword argument
+def foo(x: int, y: str) -> None:
+    pass
+z: int = 42
+foo(z, x=1, y="foo")
+

--- a/regression/python/github_3015_1_fail/test.desc
+++ b/regression/python/github_3015_1_fail/test.desc
@@ -1,0 +1,5 @@
+CORE
+main.py
+
+^SyntaxError: Multiple values for argument 'x'$
+

--- a/regression/python/github_3015_2_fail/main.py
+++ b/regression/python/github_3015_2_fail/main.py
@@ -1,0 +1,7 @@
+# Test: Unknown keyword argument (should fail)
+# Issue #3015: Passing unexpected keyword argument
+def foo(y: int, x: str) -> None:
+    pass
+z: int = 42
+foo(z, y=1, x="foo")
+

--- a/regression/python/github_3015_2_fail/test.desc
+++ b/regression/python/github_3015_2_fail/test.desc
@@ -1,0 +1,5 @@
+CORE
+main.py
+
+^SyntaxError: Multiple values for argument 'y'$
+

--- a/regression/python/github_3015_3_fail/main.py
+++ b/regression/python/github_3015_3_fail/main.py
@@ -1,0 +1,7 @@
+# Test: Unknown keyword argument (should fail)
+# Issue #3015: Passing unexpected keyword argument
+def foo(y: int, x: str) -> None:
+    pass
+z: int = 42
+foo(z )
+

--- a/regression/python/github_3015_3_fail/test.desc
+++ b/regression/python/github_3015_3_fail/test.desc
@@ -1,0 +1,5 @@
+CORE
+main.py
+
+^VERIFICATION FAILED$
+

--- a/regression/python/github_3015_4_fail/main.py
+++ b/regression/python/github_3015_4_fail/main.py
@@ -1,0 +1,7 @@
+# Test: Unknown keyword argument (should fail)
+# Issue #3015: Passing unexpected keyword argument
+def foo(x: int, y: str) -> None:
+    pass
+
+foo(z=42, x=1, y="foo")
+

--- a/regression/python/github_3015_4_fail/test.desc
+++ b/regression/python/github_3015_4_fail/test.desc
@@ -1,0 +1,5 @@
+KNOWNBUG
+main.py
+
+^SyntaxError: Unknown keyword argument: z$
+

--- a/regression/python/github_3015_4_fail/test.desc
+++ b/regression/python/github_3015_4_fail/test.desc
@@ -1,5 +1,5 @@
-KNOWNBUG
+CORE
 main.py
 
-^SyntaxError: Unknown keyword argument: z$
+^ERROR: Unknown keyword argument: z in function foo$
 

--- a/regression/python/github_3015_fail/main.py
+++ b/regression/python/github_3015_fail/main.py
@@ -1,0 +1,7 @@
+# Test: Multiple values for keyword argument (should fail)
+# Issue #3015: foo gets multiple values for argument "x"
+def foo(x: int, y: str) -> None:
+    pass
+
+foo(42, x=1, y="foo")
+

--- a/regression/python/github_3015_fail/test.desc
+++ b/regression/python/github_3015_fail/test.desc
@@ -1,0 +1,5 @@
+CORE
+main.py
+
+^SyntaxError: Multiple values for argument 'x'$
+

--- a/src/python-frontend/preprocessor.py
+++ b/src/python-frontend/preprocessor.py
@@ -1085,7 +1085,7 @@ class Preprocessor(ast.NodeTransformer):
                 raise SyntaxError(
                     f"Multiple values for argument '{expectedArgs[i]}'",
                     (self.module_name, node.lineno, node.col_offset, ""))
-        
+
         # append defaults
         for i in range(len(node.args),len(expectedArgs)):
             if expectedArgs[i] in keywords:

--- a/src/python-frontend/preprocessor.py
+++ b/src/python-frontend/preprocessor.py
@@ -1078,6 +1078,14 @@ class Preprocessor(ast.NodeTransformer):
             self.generic_visit(node)
             return node
 
+        # Check for conflicts between positional and keyword arguments
+        for i in range(len(node.args)):
+            if i < len(expectedArgs) and expectedArgs[i] in keywords:
+                # Positional argument at position i conflicts with keyword argument
+                raise SyntaxError(
+                    f"Multiple values for argument '{expectedArgs[i]}'",
+                    (self.module_name, node.lineno, node.col_offset, ""))
+        
         # append defaults
         for i in range(len(node.args),len(expectedArgs)):
             if expectedArgs[i] in keywords:

